src/components/QuicksightDashboard.tsx

```diff
@@ -0,0 +1,110 @@
+import * as React from "react";
+import {
+  QuickSightEmbeddedURLData,
+  LibrariesData,
+  LibraryData,
+} from "../interfaces";
+import { Store } from "redux";
+import { RootState } from "../store";
+import { connect } from "react-redux";
+import ActionCreator from "../actions";
+
+export interface QuicksightDashboardStateProps {
+  isFetchingLibraries?: boolean;
+  libraries?: LibraryData[];
+}
+
+export interface QuicksightDashboardDispatchProps {
+  fetchLibraries?: () => Promise<LibrariesData>;
+  fetchQuicksightEmbedUrl?: (
+    dashboardId: string,
+    ld: LibrariesData
+  ) => Promise<QuickSightEmbeddedURLData>;
+}
+
+export interface QuicksightDashboardOwnProps {
+  store?: Store<RootState>;
+  dashboardId?: string;
+}
+
+export interface QuicksightDashboardProps
+  extends QuicksightDashboardStateProps,
+    QuicksightDashboardDispatchProps,
+    QuicksightDashboardOwnProps {}
+
+export interface QuicksightDashboardState {
+  embedUrl: string;
+}
+
+class QuicksightDashboard extends React.Component<
+  QuicksightDashboardProps,
+  QuicksightDashboardState
+> {
+  constructor(props) {
+    super(props);
+    this.state = { embedUrl: null };
+  }
+  componentDidMount() {
+    if (this.state.embedUrl) {
+      return;
+    }
+    // Every time the component is mounted a fresh embed url must be fetched.
+    // TODO:  For whatever reason, the "this.props.libraries" variable was not being
+    // set when I tried to put this logic in the render method (nor did it work trying to access it in this method).
```

Comment:
It doesn't look like `libraries` is being passed as a `prop` from `QuicksightDashboardPage` l.46, so that's probably why we don't see it here. We *are* passing `dashboardId` there, but not `library`.

Reply:
Right - "libraries" is the list of libraries that is being pulled from the store - at least I'd like to be pulling it from the store.  But it doesn't get loaded. I removed the reference to "library" for now, because of the way the QS filtering is working:  it's always returning the list of libraries the user has access to.  So in the library QS view,  users can look across all libraries, any group of libraries or a single library.